### PR TITLE
go/tools/releaser: add prepare subcommand

### PR DESCRIPTION
**Pull request by jayconrod**
The prepare subcommand checks and prepares a new GitHub release.

### CODE REVIEW

**Review by robfig — APPROVED** [review]
Should have called it robojay

File: go/tools/releaser/prepare.go [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+		fmt.Fprintf(stderr, "uploading archive to mirror.bazel.build...\n")
+		if err := copyFileToMirror(ctx, arcGHURLWithoutScheme, arcName); err != nil {
+			return err
+		}
+	}
+
+	// Create or update the GitHub release.
+	if release == nil {
+		fmt.Fprintf(stderr, "creating draft release...\n")
+		prerelease := true
```
Comment (robfig) [inline go/tools/releaser/prepare.go:195]: Why prerelease?

Related issue:
https://github.com/bazelbuild/rules_go/issues/2899

Comment (jayconrod) [reply go/tools/releaser/prepare.go]: This tool will no longer set the prerelease bit. I don't think there was any particular reason for it, but from #2899, it sounds like it was causing issues. The version number should express whether something is prerelease or not.

**Review by robfig — APPROVED** [review]
LGTM, thanks! Good to go?

**Comment by jayconrod** [comment]
Yep. I've also updated instructions at https://github.com/bazelbuild/rules_go/wiki/Updating-dependencies#how-to-update.

**Pull request merged**